src/main/java/org/openstreetmap/atlas/checks/validation/intersections/EdgeCrossingEdgeCheck.java

```diff
@@ -188,7 +181,7 @@ public boolean validCheckForObject(final AtlasObject object)
     private boolean isValidCrossingEdge(final AtlasObject object)
     {
         if (Edge.isMasterEdgeIdentifier(object.getIdentifier())
-                && !TagPredicates.IS_AREA.test(object) && !this.isFlagged(object.getIdentifier()))
```

Comment:
This does prevent an Edge from being inconsistently flagged, but it also creates a lot of duplication in the flags. Ideally we would want to avoid both those things.

Reply:
I understand your concern. But, I thought we wanted to flag every edge unlike the node in DuplicateNodeCheck. Please let me know if you are suggesting to fix this bug in similar fashion of duplicate node check.

Reply:
Just a suggestion, we could do something like grabbing all ```collectedEdges``` and flagging all the edges except for the one with the lowest OSM ID. This should ensure we have consistent flagging per set of collected edges, and flags don't depend on which edge is the starting edge.